app/controllers/concerns/keycloak_utils.rb

```diff
@@ -9,18 +9,25 @@ module KeycloakUtils
     optional(:last_name).filled(:string)
     optional(:company).filled(:string)
     optional(:department).filled(:string)
-    required(:role).filled(:string)
+    required(:resource_access).value(:hash) do
+      required(:homs).value(:hash) do
+        required(:roles).filled(:array, size?: 1) do
+          each(included_in?: User.roles_list)
+        end
+      end
+    end
   end
 
   def validate_user_data(user_data)
     UserDataSchema.call(
-      email:      user_data[:email],
-      name:       user_data[:name],
-      last_name:  user_data[:last_name],
-      company:    user_data[:company],
-      department: user_data[:department],
-      role:       user_data[:role]
+      email:           user_data[:email],
+      name:            user_data[:name],
+      last_name:       user_data[:last_name],
+      company:         user_data[:company],
+      department:      user_data[:department],
+      resource_access: user_data[:resource_access]
     )
                   .to_monad
+                  .fmap { |r| r.to_h.merge(role: r.to_h[:resource_access][:homs][:roles].pop) }
```

Comment:
1. `pop` is a mutating method so avoid using it as getter
1. use `dig` on ruby hash to fetch deeply nested key value
1. I would use the [dry-transformer](https://dry-rb.org/gems/dry-transformer/0.1/) for this kind of things